File: 02-LINKED-LISTS/number-of-students-unable-to-eat-lunch.php
<?php

# https://leetcode.com/problems/number-of-students-unable-to-eat-lunch/
class Solution
{
    function countStudents(array $students, array $sandwiches): int
    {
        $requeue = 0;
        while ($students && $requeue < count($students)) {
            $student = array_shift($students);
            if ($student == $sandwiches[0]) {
                array_shift($sandwiches);
                $requeue = 0;
                continue;
            }
            $students[] = $student;
            $requeue++;
        }

        return $requeue;
    }
}

File: 02-LINKED-LISTS/number-of-students-unable-to-eat-lunch.py
from typing import *


# https://leetcode.com/problems/number-of-students-unable-to-eat-lunch/
class Solution:
    def countStudents(self, students: List[int], sandwiches: List[int]) -> int:
        requeue = 0
        while students and requeue < len(students):
            student = students.pop(0)
            if student == sandwiches[0]:
                sandwiches.pop(0)
                requeue = 0
                continue
            students.append(student)
            requeue += 1

        return requeue
